exporter/logstashexporter/sync.go

```diff
@@ -204,16 +209,27 @@ func (c *syncClient) publishWindowed(events []plog.LogRecord) (int, error) {
 }
 
 func (c *syncClient) sendEvents(events []plog.LogRecord) (int, error) {
-	window := make([]interface{}, len(events))
+	window := make([]interface{}, 0, len(events))
+	//TODO: needs more tests and definitions regarding non-beats events actions
 	for i := range events {
-		record := events[i]
-		//TODO: Testing purpose, it should deserialize from LogRecord.Body, send it as-it-is?
-		event := beat.Event{
-			Timestamp: record.Timestamp().AsTime(),
-			Meta:      map[string]interface{}{},
-			Fields:    record.Attributes().AsRaw(),
+		logRecord := events[i]
+		logRecordBody, ok := newLogRecordBody(&logRecord)
+		if !ok {
+			return 0, consumererror.NewPermanent(errors.New("invalid beats event body"))
 		}
-		window[i] = &event
+
+		metadata := extractEventMetadata(logRecordBody)
+		if !isBeatsEvent(metadata) {
+			return 0, consumererror.NewPermanent(errors.New("received a non-beats event"))
```

Comment:
I'd suggest leaving a TODO here to explore adding a metric

```suggestion
			// TODO: Consider producing a metric instead to avoid heavy/per-event logging
			return 0, consumererror.NewPermanent(errors.New("received a non-beats event"))
```

Reply:
The original code sends the metrics here: https://github.com/andsel/opentelemetry-collector-components/pull/5/files/1e757668b992d6be58d155969d01999a531442e7#diff-0a74fb7652a3cbc32b4df3a88f84a532acaee1d4c7251a7e0cbb42beb1758ae4R177, if will need to implement an Observer for the collector and replace the current noop impl.